lib/screens/dashboard/view/dashboard_screen.dart

```diff
@@ -144,22 +156,49 @@ class DashboardScreenState extends State<DashboardScreen>
           : Column(
               crossAxisAlignment: CrossAxisAlignment.start,
               children: [
-                StepBar(
-                    actionable: true,
-                    selectedItem: dashboardSteps[selectedItem]!,
-                    listItems: dashboardSteps.values.toList(),
-                    onChanged: (value) => {
-                          setState(() => selectedItem =
-                              getEnumFromString(dashboardSteps, value ?? '')!
-                                  as DashboardStepBar)
-                        }),
+                _configView(),
                 SizedBox(height: 16),
-                buildMainDashboardContent(selectedItem, theme)
+                buildMainDashboardContent(theme)
               ],
             );
     });
   }
 
+  StepBar get stepBar {
+    if (_stepBar == null) {
+      _stepBar = StepBar(
+          key: _stepBarKey,
+          steps: _localizedDashboardViewSteps.values.toList(),
+          actionable: false,
+          initialItem: _localizedDashboardViewSteps.values.first,
+          onChanged: (item) => {
+            setState(() {
+              _stepBarKey.currentState!.setState(() {
+                _stepBarKey.currentState!.selectedItem = item!;
+              });
+            })
+          });
+    }
+    return _stepBar!;
+  }
+
+  Widget _configView() {
+    if (_stepBar == null) {
+      _stepBar = StepBar(
+          key: _stepBarKey,
+          actionable: true,
+          steps: _localizedDashboardViewSteps.values.toList(),
+          initialItem: null,
+          onChanged: (item) => {
+            scrollController.jumpTo(0.0),
+            setState(
+                  () => _stepBarKey.currentState!.selectedItem = item!,
```

Comment:
Why it is necessary to use a key to update the selectedItem in the `StepBar` component?